falcon/response_helpers.py

```diff
@@ -51,12 +51,16 @@ def format_range(value):
 
     Args:
         value: ``tuple`` passed to `req.range`
-
     """
 
     # PERF: Concatenation is faster than % string formatting as well
     #       as ''.join() in this case.
-    return ('bytes ' +
+    if len(value) == 4:
+        unit = value[3]
+    else:
+        unit = 'bytes'
```

Comment:
PERF: Probably not significant, but if you'd like to save an extra concatenation below, you could include the space in the literal string in L61 (i.e., `'bytes '`) and then append a space to the custom unit type in L59 (i.e., `value[3] + ' '`). That way, you only pay the penalty when using a custom unit type.